### PR TITLE
 Add a checkbox on autoconvert for strings of consistent encodings

### PR DESCRIPTION
**Pull request by sjenks**
Do not merge yet: this needs the unpublished version of puzzle-lib

### CODE REVIEW

**Comment by kfarnung** [comment]
Published `puzzle-lib@0.4.0` to NPM:  https://www.npmjs.com/package/puzzle-lib

File: src/Conversion/AutoConversion/AutoConvertStream.tsx [hunk 1]
```diff
@@ -47,6 +51,11 @@ class AutoConvertStream extends LocalStorageComponent<Props, State, SavedState>
             <Button onClick={(event: React.MouseEvent<Button>) => this.onClearClick(event)}>Clear</Button>
           </ButtonGroup>
         </ButtonToolbar>
+        <input 
+          type="checkbox"
+          onClick={(event: React.MouseEvent<HTMLInputElement>) => this.toggleHomogenous(event)}
```
Comment (kfarnung) [inline src/Conversion/AutoConversion/AutoConvertStream.tsx]: You'll want to bind `this.state.homogeneous` to the `checked` property of the input element.  Then the state will be restored correctly.

File: src/Conversion/AutoConversion/AutoConvertStream.tsx [hunk 2]
```diff
@@ -47,6 +51,11 @@ class AutoConvertStream extends LocalStorageComponent<Props, State, SavedState>
             <Button onClick={(event: React.MouseEvent<Button>) => this.onClearClick(event)}>Clear</Button>
           </ButtonGroup>
         </ButtonToolbar>
+        <input 
+          type="checkbox"
```
Comment (kfarnung) [inline src/Conversion/AutoConversion/AutoConvertStream.tsx]: Have you tried a `Button` and binding to the `active` state?

https://react-bootstrap.github.io/components/buttons/#buttons-active

Comment (kfarnung) [reply src/Conversion/AutoConversion/AutoConvertStream.tsx]: Then you can put it in the `ButtonToolbar`

File: src/Conversion/AutoConversion/AutoConvertStream.tsx [hunk 3]
```diff
@@ -88,10 +100,13 @@ class AutoConvertStream extends LocalStorageComponent<Props, State, SavedState>
     this.updateState();
   }
 
+  private toggleHomogenous(event: React.MouseEvent<HTMLInputElement>) {
```
Comment (kfarnung) [inline src/Conversion/AutoConversion/AutoConvertStream.tsx]: nit: typo `toggleHomogeneous`

File: src/Conversion/AutoConversion/AutoConvertStream.tsx (hunk 1 above)
Comment (sjenks) [reply src/Conversion/AutoConversion/AutoConvertStream.tsx]: Maybe I'm doing it wrong, but I'm getting:

"Binds are forbidden in JSX attributes due to their rendering performance impact (jsx-no-bind)" - TSLint

File: src/Conversion/AutoConversion/AutoConvertStream.tsx (hunk 3 above)
Comment (sjenks) [reply src/Conversion/AutoConversion/AutoConvertStream.tsx]: Fixed

File: src/Conversion/AutoConversion/AutoConvertStream.tsx (hunk 1 above)
Comment (kfarnung) [reply src/Conversion/AutoConversion/AutoConvertStream.tsx]: It should just be like:

```
<input checked={this.state.homogeneous} />
```

Comment (kfarnung) [reply src/Conversion/AutoConversion/AutoConvertStream.tsx]: You may also want to take a look at my semaphore UI usage of checkboxes, I listen to onChange and get the value of checked.

File: src/Conversion/AutoConversion/AutoConvertStream.tsx (hunk 2 above)
Comment (kfarnung) [reply src/Conversion/AutoConversion/AutoConvertStream.tsx]: Actually, it looks like you want:  https://react-bootstrap.github.io/components/button-group/

File: src/Conversion/AutoConversion/AutoConvertStream.css [hunk 4]
```diff
@@ -2,10 +2,6 @@
   margin-bottom: 10px;
 }
 
-.AutoConvertStream-commands .btn-group {
```
Comment (kfarnung) [inline src/Conversion/AutoConversion/AutoConvertStream.css]: You probably still want this?

File: src/Conversion/AutoConversion/AutoConvertStream.tsx [hunk 5]
```diff
@@ -46,19 +46,23 @@ class AutoConvertStream extends LocalStorageComponent<Props, State, SavedState>
           placeholder="Text"
           value={this.state.text}
         />
-        <ButtonToolbar className="AutoConvertStream-commands">
```
Comment (kfarnung) [inline src/Conversion/AutoConversion/AutoConvertStream.tsx]: You likely want to keep the button toolbar

**Pull request merged**